lib/Elevate/Blockers/Leapp.pm

```diff
@@ -29,29 +29,51 @@ sub check ($self) {
 
     $self->cpev->leapp->install();
 
-    $self->cpev->leapp->preupgrade();
+    my $out = $self->cpev->leapp->preupgrade();
 
-    my $blockers = $self->cpev->leapp->search_report_file_for_blockers(
+    # A return code of zero indicates that no inhibitors
+    # or fatal errors have been found.
+    return if ( $out->{status} == 0 );
+
+    # Leapp will generate a JSON report file which contains any
+    # inhibitors found. Find any reported inhibitors but exclude ones
+    # that we know about and will fix before the upgrade.
+    # (Inhibitors will also be reported in stdout in a block
+    # labeled "UPGRADE INHIBITORS"; but more complete info is reported
+    # in the JSON report file.)
+
+    my $inhibitors = $self->cpev->leapp->search_report_file_for_inhibitors(
         qw(
           check_installed_devel_kernels
           cl_mysql_repository_setup
           verify_check_results
         )
     );
 
-    foreach my $blocker (@$blockers) {
-        my $message = $blocker->{title} . "\n";
-        $message .= $blocker->{summary} . "\n";
-        if ( $blocker->{hint} ) {
-            $message .= "Possible resolution: " . $blocker->{hint} . "\n";
+    foreach my $inhibitor (@$inhibitors) {
+        my $message = $inhibitor->{title} . "\n";
+        $message .= $inhibitor->{summary} . "\n";
+        if ( $inhibitor->{hint} ) {
+            $message .= "Possible resolution: " . $inhibitor->{hint} . "\n";
         }
-        if ( $blocker->{command} ) {
-            $message .= "Consider running:" . "\n" . $blocker->{command} . "\n";
+        if ( $inhibitor->{command} ) {
+            $message .= "Consider running:" . "\n" . $inhibitor->{command} . "\n";
         }
 
         $self->has_blocker($message);
     }
 
+    # Fatal errors will NOT be flagged as inhibitors in the
+    # leapp reports.  So it is NOT possible to distinguish them from
+    # any non-fatal conditions reported there.  So, we need to fish
+    # them from stdout.
```

Comment:
In our experience, a fatal error caused leapp to halt immediately and not look for any more potential inhibitors.  So, we've only seen one or the other.  However, that is not guaranteed.  Hypothetically, it could encounter a fatal error after finding an inhibitor thus having both.  There will be no redundancies since inhibitors do not appear in the ERROR block in the output.  They show up in a block labeled "UPGRADE INHIBITORS" instead.

Reply:
That makes sense.  Consider this question resolved.